### PR TITLE
docs: Update note about presence of boot settings to k8s-1.23 and beyond

### PR DESCRIPTION
**Pull request by markusboehme**
**Issue number:** n/a

**Description of changes:**

Update the readme to add a note about the presence of boot settings. The settings model includes the boot settings on all k8s-1.23 variants now. Of the non-devel variants, metal-k8s-1.22 was first and deserves a special mention.

**Testing done:**

```
$ git grep 'boot: BootSettings'
sources/models/src/aws-k8s-1.23-nvidia/mod.rs:    boot: BootSettings,
sources/models/src/aws-k8s-1.23/mod.rs:    boot: BootSettings,
sources/models/src/metal-dev/mod.rs:    boot: BootSettings,
sources/models/src/metal-k8s-1.22/mod.rs:    boot: BootSettings,
sources/models/src/metal-k8s-1.23/mod.rs:    boot: BootSettings,
sources/models/src/vmware-k8s-1.23/mod.rs:    boot: BootSettings,
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by foersleo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -699,7 +699,7 @@ Here are the metrics settings:
 
 #### Boot-related settings
 
-*Please note that boot settings only exist for bare-metal variants at the moment*
+*Please note that boot settings only exist for the metal-k8s-1.22 variant and the k8s-1.23 (or above) variants.*
```
Comment (etungsten) [inline README.md]: `metal-k8s-1.21`'s `mod.rs` is symlinked to `metal-k8s-1.22`'s `mod.rs`
```
$ ls -al sources/models/src/metal-*
lrwxrwxrwx 1 etung domain^users   16 Jun  7 20:09 sources/models/src/metal-k8s-1.21 -> ./metal-k8s-1.22
```

It's the `metal-dev` variant as well (or just all metal variants). And we're planning to have the boot settings for all future variants (not limited to `*-k8s-*`). So maybe?:
```suggestion
*Please note that boot settings currently only exist for the baremetal variants and *-k8s-1.23 variants. Boot settings will be added to any future variant introduced after Bottlerocket v1.8.0.
```

**Review by etungsten — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (markusboehme) [reply README.md]: Thanks, your suggestion is much clearer about expectations for future variants!

**Comment by markusboehme** [comment]
Force push adopts Erikson's suggested wording.

**Review by zmrow — APPROVED** [review]
Thanks!

:clinking_glasses:

**Review by etungsten — APPROVED** [review]

**Review by mchaker — APPROVED** [review]

**Pull request merged**